### PR TITLE
SAPT-D!

### PR DESCRIPTION
**Pull request by dsirianni**
## Description

Adds additive dispersion correction to SAPT (pub upcoming from CDS group) to engine, and enables return of atomic pairwise dispersion from DFTD3.

## Changelog description

Adds SAPT-D & pairwise DFTD3 analysis.

## Status

- [x] Code base linted
- [x] Ready to go

### CODE REVIEW

File: qcengine/programs/dftd3.py [hunk 1]
```diff
@@ -143,6 +143,10 @@ def build_input(
         if input_model.extras["info"]["dashlevel"] == "atmgr":
             command.append("-abc")
 
+        # Append `-anal` for pairwise atomic analysis
+        if input_model.driver != "gradient":
```
Comment (dgasmith) [inline qcengine/programs/dftd3.py]: @loriab Seems like this should be an optional flag in keywords?

File: qcengine/programs/dftd3.py [hunk 2]
```diff
@@ -187,6 +191,22 @@ def parse_output(self, outfiles: Dict[str, str], input_model: "AtomicInput") ->
...
+                # Iterate over block
+                start = stdout.splitlines().index(ln) + 2
+                for l in stdout.splitlines()[start:]:
+                    data = l.replace("-", " -").split()
+                    # print(data)
+                    if len(data) == 0:
+                        break
+                    atom1 = int(data[0]) - 1
+                    atom2 = int(data[1]) - 1
+                    Edisp = Decimal(data[-1])
```
Comment (dgasmith) [inline qcengine/programs/dftd3.py]: Not sure we want to store this data as Decimal, these things are pretty explicitly floats.

File: qcengine/programs/dftd3.py [hunk 3]
```diff
@@ -279,8 +299,10 @@ def parse_output(self, outfiles: Dict[str, str], input_model: "AtomicInput") ->
         }
         output_data["extras"]["local_keywords"] = input_model.extras["info"]
         output_data["extras"]["qcvars"] = calcinfo
-
+        if input_model.driver != "gradient":
+            output_data["extras"]["qcvars"]["PAIRWISE DISPERSION CORRECTION ANALYSIS"] = D3pairs
```
Comment (dgasmith) [inline qcengine/programs/dftd3.py]: We need to check this term explicitly, perhaps the norm of the values?

Comment (loriab) [reply qcengine/programs/dftd3.py]: yes, that's the focus of this afternoon's session :-)

File: qcengine/programs/dftd3.py (hunk 2 above)
Comment (loriab) [reply qcengine/programs/dftd3.py]: `Decimal` here is so as to not lose the info of how many digits of float we are getting back from the program.

File: qcengine/programs/empirical_dispersion_resources.py [hunk 4]
```diff
@@ -649,7 +649,7 @@
             "pbe": {"params": {"s6": 1.000, "s8": 0.358940, "a1": 0.012092, "a2": 5.938951}},
             "pbe0": {"params": {"s6": 1.000, "s8": 0.528823, "a1": 0.007912, "a2": 6.162326}},
             "wpbe": {"params": {"s6": 1.000, "s8": 0.906564, "a1": 0.563761, "a2": 3.593680}},  # formerly lcwpbe
-            "sapt0": {"params": {"s6": 1.000, "s8": 1.409027, "a1": 0.078803, "a2": 4.553948}},  # experimental
+            "sapt0": {"params": {"s6": 1.000, "s8": 1.03220439, "a1": 0.00704073, "a2": 6.11845609}},  # experimental
```
Comment (loriab) [inline qcengine/programs/empirical_dispersion_resources.py:652]: how's the refitting going, @dsirianni? note that 4-6 sig figs is plenty, so perhaps the procedure is optimize, truncate, recompute to make sure we're using the same tidy parameter values here and in paper.

**Comment by loriab** [comment]
superseded by #282

**Pull request closed without merging**